mmvb_backend/benchmarking_sessions/api/views.py

```diff
@@ -0,0 +1,124 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.schemas.openapi import AutoSchema
+from rest_framework.viewsets import ModelViewSet
+
+from benchmarking_sessions.api.serializers import (
+    BenchmarkingSessionResultsSerializer,
+    BenchmarkingSessionSerializer,
+)
+from benchmarking_sessions.models import (
+    BenchmarkingSession,
+    BenchmarkingStepStatus,
+)
+from benchmarking_sessions.tasks import run_benchmark
+
+
+# todo: properly document endpoints
+class BenchmarkingSessionViewSet(ModelViewSet):
+    schema = AutoSchema(tags=["BenchmarkingSessions",])
+    serializer_class = BenchmarkingSessionSerializer
+
+    def get_queryset(self):
+        return BenchmarkingSession.objects.all()
+
+    # todo: generate 404
+    @action(methods=["post"], detail=True, url_path="run")
+    def run_benchmark(self, request, *args, **kwargs):
+        benchmarking_session = get_object_or_404(
+            BenchmarkingSession, id=kwargs["pk"]
+        )
+        task = run_benchmark.delay(benchmarking_session.id)
+        benchmarking_session.task_id = task.id
+        benchmarking_session.save()
+        return Response(
+            f"/benchmarking-sessions/{benchmarking_session.id}/status",
```

Comment:
I'll check it out locally to see how the response is returned in this case but I'm personally not a fan of  the content not being attached to a name there.
I would return something like:
```
return Response(
    {"status_url": f"/benchmarking-sessions/{benchmarking_session.id}/status"},
    status=status.HTTP_202_ACCEPTED,
)
```